app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -489,6 +787,19 @@ public void doNarrow(NarrowFilter filter) {
     }
 
     @Override
+    public void onNarrowFillSendBox(Message message) {
+        if(message.getType() == MessageType.PRIVATE_MESSAGE){
+            switchToPrivate();
+            streamActv.setText(message.getReplyTo(app));
+            messageEt.requestFocus();
+        } else {
+            switchToStream();
+            streamActv.setText(message.getStream().getName());
+            topicActv.setText(message.getSubject());
+            if (message.getSubject().equals("")) topicActv.requestFocus();
```

Comment:
Subject shouldn't be null here, as in the else case this is for stream's anyways I'll then also replace the code! :+1: